lib/jerboa/format/stun/bare.ex

```diff
@@ -0,0 +1,129 @@
+defmodule Jerboa.Format.STUN.Bare do
+  @moduledoc false
+
+  alias Jerboa.Format.STUN
+
+  defstruct [:t_id, :class, :method, attrs: [], raw: <<>>]
+
+  @stun_magic STUN.magic
+
+  @type bare_attr :: {type :: non_neg_integer, value :: binary}
+
+  @type t :: %__MODULE__{
+      t_id: non_neg_integer,
+     class: STUN.class,
+    method: non_neg_integer,
+     attrs: [bare_attr],
+       raw: binary
+  }
+
+  @spec decode(packet :: binary) :: {:ok, t} | {:ok, t, binary} | {:error, reason :: term}
+  def decode(packet) do
+    with {:ok, header} <- validate_header_length(packet),
+                   :ok <- validate_first_two_bits(header),
```

Comment:
Does it make sense to create one function `validate_header` which will return `{ok, :header}` only if `validate_header_lengt`, `validate_first_to_bits` and `validate_stun_magic` pass? This will probably make the `with` block shorter and simpler to read and understand.